lib/md051.js

```diff
@@ -0,0 +1,50 @@
+// @ts-check
+
+"use strict";
+
+const { filterTokens, addError, forEachHeading } = require("../helpers");
+
+/**
+ * Slugifies strings.
```

Comment:
Something like "Converts a Markdown heading into an HTML fragment according to the rules used by GitHub", maybe.